rviz_default_plugins/src/rviz_default_plugins/displays/map/map_display.cpp

```diff
@@ -360,10 +360,10 @@ void MapDisplay::updateMapDataInMemory(
   const map_msgs::msg::OccupancyGridUpdate::ConstSharedPtr update)
 {
   for (size_t y = 0; y < update->height; y++) {
-    std::copy(
-      update->data.begin(),
-      update->data.begin() + update->width,
-      current_map_.data.begin() + (update->y + y) * current_map_.info.width + update->x);
+    for (size_t x = 0; x < update->width; x++) {
+      current_map_.data[(update->x + x) + (update->y + y) * current_map_.info.width] =
+            update->data[x + y * update->width];
+    }
```

Comment:
So I absolutely agree that this is a bug; it looks like it was mis-ported from RViz originally.  For reference, here is a link to the original RViz code that does the same thing: https://github.com/ros-visualization/rviz/blob/1f622b8c95b8e188841b5505db2f97394d3e9c6c/src/rviz/default_plugin/map_display.cpp#L582-L587 .

That said, while this change will work, it will likely be a lot slower.  Here's my suggested fix which should fix the issue and be the same speed:


```suggestion
    for (size_t x = 0; x < update->width; x++) {
      auto offset = update->data.begin() + y * update->width;
      std::copy(
        offset,
        offset + update->width,
        current_map_.data.begin() + (update->y + y) * current_map_.info.width + update->x);
    }
```

Reply:
Yep, it works just fine.

Reply:
Note that you have to swap y for x.

Reply:
Note that you will have to change the for loop to use y instead of x.

Reply:
Did it, please review it :)

Reply:
> Note that you will have to change the for loop to use y instead of x.

Oops, you are totally correct.  Thanks for pointing that out.